docling_eval/evaluators/ocr/geometry_utils.py

```diff
@@ -0,0 +1,147 @@
+from typing import List, Tuple
+
+from docling_eval.evaluators.ocr.evaluation_models import Location, Word
+
+
+def create_polygon_from_rect(location: Location) -> List[List[float]]:
+    return [
+        [location.left, location.top],
+        [location.left + location.width, location.top],
+        [location.left + location.width, location.top + location.height],
+        [location.left, location.top + location.height],
+    ]
+
+
+def convert_locations_to_float(words: List[Word]) -> None:
+    for word_item in words:
+        loc: Location = word_item.location
+        loc.top = float(loc.top)
+        loc.left = float(loc.left)
+        loc.width = float(loc.width)
+        loc.height = float(loc.height)
+        loc.right = loc.left + loc.width
+        loc.bottom = loc.top + loc.height
+    return
+
+
+def get_x_axis_overlap(rect1: Location, rect2: Location) -> float:
+    x_overlap: float = max(
+        0.0, min(rect1.right, rect2.right) - max(rect1.left, rect2.left)
+    )
+    return x_overlap
+
+
+def get_y_axis_overlap(rect1: Location, rect2: Location) -> float:
+    y_overlap: float = max(
+        0.0, min(rect1.bottom, rect2.bottom) - max(rect1.top, rect2.top)
+    )
+    return y_overlap
+
+
+def get_intersection_area(rect1: Location, rect2: Location) -> float:
```

Comment:
Let's migrate these functions to docling-core to the `BoundingBox` (https://github.com/docling-project/docling-core/blob/main/docling_core/types/doc/base.py#L35) as class methods